pkg/util/systemd_action.go

```diff
@@ -46,3 +47,30 @@ func StopUnit(d *dbus.Conn, unitName string) error {
 	glog.Infof("Stopping systemd unit: %s ...", unitName)
 	return executeSystemdAction(unitName, d.StopUnit)
 }
+
+func GetUnitStates(d *dbus.Conn, unitNames []string) ([]dbus.UnitStatus, error) {
+	var units []dbus.UnitStatus
+
+	allUnits, err := d.ListUnits()
+	if err != nil {
+		glog.Errorf("Cannot ListUnits: %v", err)
+		return nil, err
+	}
+	intersect := make(map[string]dbus.UnitStatus)
+	for _, elt := range allUnits {
```

Comment:
Would it make sense to check the jobID of a unit collected and make sure you are not adding the same unit with a different name several times. Because "units may be known by multiple names at the same time, and hence there might be more unit names loaded than actual units behind them." (From the doc)

Reply:
I added this safety to the current code path.

I'm not sure where we could have this issue but it's easy to catch this.